rs/interfaces/src/crypto/vetkd.rs

```diff
@@ -0,0 +1,32 @@
+use ic_types::crypto::vetkd::{
+    VedKdKeyShareCreationError, VetKdArgs, VetKdEncryptedKey, VetKdEncryptedKeyShare,
+    VetKdKeyShareCombinationError, VetKdKeyShareVerificationError, VetKdKeyVerificationError,
+};
+use ic_types::NodeId;
+use std::collections::BTreeMap;
+
+pub trait VetKdProtocol {
+    fn create_encrypted_key_share(
+        &self,
+        args: VetKdArgs,
```

Comment:
Looks like it is still passed by value in `create_encrypted_key_share`, is that intentional?

Reply:
Yes, mainly because of [C-CALLER-CONTROL](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control), given that we'll send most of the contained data via RPC to the vault, so we need to own the bytes to put them on the wire. But it's not a clear cut: see https://github.com/dfinity/ic/pull/2569#discussion_r1840297054.